pkcs8/src/attributes.rs

```diff
@@ -1,31 +1,70 @@
+//! `OneAsymmetricKey` attributes.
+
 use core::convert::TryFrom;
 
-use der::{Any, Encodable, Encoder, Error, Length, Result, Tag, Tagged};
+use der::{Any, Encodable, Encoder, Error, Header, Length, Result, Tag, Tagged};
+
+/// Attributes as defined in [RFC 5958 Section 2]:
+///
+/// > attributes is OPTIONAL.  It contains information corresponding to
+/// > the public key (e.g., certificates).  The attributes field uses the
+/// > class `ATTRIBUTE` which is restricted by the
+/// > `OneAsymmetricKeyAttributes` information object set.
+/// > `OneAsymmetricKeyAttributes` is an open ended set in this document.
+/// > Others documents can constrain these values.  Attributes from
+/// > RFC2985 MAY be supported.
+///
+/// Attributes have the following ASN.1 schema:
+///
+/// ```text
+/// Attributes ::= SET OF Attribute { { OneAsymmetricKeyAttributes } }
+/// ```
+///
+/// [RFC 5958 Section 2]: https://datatracker.ietf.org/doc/html/rfc5958
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub struct Attributes<'a> {
+    /// Inner ASN.1 value (i.e. `SET OF Attribute`).
+    // TODO(tarcieri): decode this as a `SetOf`?
```

Comment:
it's very unlikely that attributes would exist in the wild, i'd say it's okay to leave this as undefined as it is here as a semi-scream-test, so to see if someone has trouble decoding a particular document because it has attributes, so it's okay for now.

Reply:
I think attributes may actually be used to store the certificate in the context of PKCS#12 (which is an extension of PKCS#8v2), but I'm uncertain.

Reply:
nw nw